File: src/main/java/com/tabnine/state/UserState.java
package com.tabnine.state;

import static com.tabnine.general.DependencyContainer.instanceOfBinaryRequestFacade;

import com.tabnine.binary.BinaryRequestFacade;
import com.tabnine.binary.requests.config.StateRequest;
import com.tabnine.binary.requests.config.StateResponse;

public class UserState {
  public static UserState userState;
  private final SuggestionHintState suggestionHintState;

  public static void init() {
    if (userState == null) {
      userState = new UserState();
    }
  }

  public static UserState getInstance() {
    init();
    return userState;
  }

  private UserState() {
    BinaryRequestFacade binaryRequestFacade = instanceOfBinaryRequestFacade();
    StateResponse stateResponse = binaryRequestFacade.executeRequest(new StateRequest());
    suggestionHintState = new SuggestionHintState(stateResponse.getInstallationTime());
  }

  public SuggestionHintState getSuggestionHintState() {
    return suggestionHintState;
  }
}

File: src/main/java/com/tabnine/state/UserState.kt
package com.tabnine.state

import com.tabnine.binary.requests.config.StateRequest
import com.tabnine.general.DependencyContainer
import com.tabnine.general.ServiceLevel

class UserState private constructor() {
    val suggestionHintState: SuggestionHintState
    val serviceLevel: ServiceLevel?

    init {
        val binaryRequestFacade = DependencyContainer.instanceOfBinaryRequestFacade()
        val stateResponse = binaryRequestFacade.executeRequest(StateRequest())
        suggestionHintState = SuggestionHintState(stateResponse?.installationTime)
        serviceLevel = stateResponse?.serviceLevel
    }

    companion object {
        var userState: UserState? = null

        @JvmStatic
        fun init() {
            if (userState == null) {
                userState = UserState()
            }
        }

        @JvmStatic
        val instance: UserState
            get() {
                init()
                return userState!!
            }
    }
}
